File: OOP/4_single_inheritance.cpp
#include <iostream>
#include <string>
using namespace std;

// Parent class A
class A {
public:
    string name;

    // Constructor for class A
    A(string name) {
        this->name = name;
    }

    // Display method to print the name
    void display() {
        cout << name << endl;
    }
};

// Child class B inheriting from class A
class B : public A {
public:
    // Constructor for class B
    B(string name) : A(name) {
        // Calling parent class constructor using initializer list
    }
};

int main() {
    
    B obj("Salim");   //  Creating an object of class B with name "Salim"
    obj.display();

    return 0;
}

File: OOP/4_single_inheritance.py

class A:
    def __init__(self,name) -> None:
        self.name = name

    def display(self):
        print(self.name)


class B(A):
    def __init__(self, name) -> None:
        super().__init__(name)

obj = B("Salim")
obj.display()



# practise

# class C:
#     def __init__(self,age) -> None:
#         self.age = age

#     def display(self):
#         print(self.age)
    
# class D(C):
#     def __init__(self, age) -> None:
#         super().__init__(age)  # super er self.age e 27 set hosse

#     def display(self):
#         super().display()  # age super call hosse
#         print("I am child class and I am also 27")

# obj = D(27)
# obj.display()

    